constraintlayout/core/src/main/java/androidx/constraintlayout/core/dsl/Constraint.java

```diff
@@ -38,9 +43,39 @@ public class Anchor {
             mSide = side;
         }
 
+        public String getId() {
+            return mId;
+        }
+
         Constraint getParent() {
             return Constraint.this;
         }
+
+        @Override
+        public String toString() {
+            StringBuilder ret = new StringBuilder("[");
+
+            if (mConnection != null) {
+                ret.append("'").append(mConnection.getId()).append("',")
```

Comment:
For efficiency 
have a build(StringBuilder builder)  method.
that you use

Reply:
Done!